docs: Small docs updates, peering and K8s CNI (#15702)
hashicorp/consul: website/content/docs/k8s/installation/install.mdx

The following table describes the available CNI plugin options:

| Option     | Description | Default       |
| ---------- | ----------- | ------------- |
| `cni.enabled` | Boolean value that enables or disables the CNI plugin. If `true`, the plugin is responsible for redirecting traffic in the service mesh. If `false`, redirection is handled by the `connect-inject init` container. | `false` |
| `cni.logLevel` | String value that specifies the log level for the installer and plugin. You can specify the following values: `info`, `debug`, `error`. | `info` |

> | `cni.namespace` | Set the namespace to install the CNI plugin into. Overrides global namespace settings for CNI resources, i.e. `kube-system` | namespace used for `consul-k8s` install, i.e. `consul` |

Reviewer: ```suggestion
| `cni.namespace` | Specifies the namespace where the CNI plugin is installed. This configuration overrides the `kube-system`global namespace settings for CNI resources. | `consul` or namespace used for `consul-k8s` installation  |
```
Trying to make this information more precise. Also, using latinate abbreviations (i.e., e.g., etc.) is a common style guide violation, which is something I just learned about myself, probably because they are so imprecise.
